### DISCUSSION TITLE
I am trying to change the tab basically using valtio. but it goes into renderloop

### DISCUSSION THREAD

**Discussion by afkcodes** [pmndrs/valtio#789] [Q&A]
I have tried a very basic implementation. whenever i change the tab it goes into render loop. useState works fine though.

Below is how i am updating tab
![Screenshot 2023-09-18 at 8 18 37 PM](https://github.com/pmndrs/valtio/assets/17390257/c22d78ce-a7e0-4a29-8376-720cabb68edf)

Here is the Valtio state
![Screenshot 2023-09-18 at 8 17 58 PM](https://github.com/pmndrs/valtio/assets/17390257/29255fc3-5d29-4085-a890-60bba3db0122)

Here is how Bottom Tab is being used
![Screenshot 2023-09-18 at 9 28 05 PM](https://github.com/pmndrs/valtio/assets/17390257/c0600e6b-71b8-410a-b2be-b00d717bff18)

Please let me know if i am doing anything wrong here, thanks

**Comment by dai-shi** [comment]
Hmm, it looks good to me at a glance. Would you like to try creating a minimal reproduction?

**Reply by afkcodes** [reply]
Closing this as found out this is nothing do with valtio, its basically the react thing where it batches the state updates. using sync:true worked fine.

@dai-shi  can you explain a bit the usage of `{sync:true}` as i was not very clear what is happening, a little heads up will be great.

**Reply by dai-shi** [reply]
Does this comment help? https://github.com/pmndrs/valtio/issues/687#issuecomment-1734936679
